manager/apis/app/v1alpha1/fybrikapplication_types.go

```diff
@@ -155,25 +155,34 @@ type DatasetDetails struct {
 	Details serde.Arbitrary `json:"details,omitempty"`
 }
 
-// FybrikApplicationStatus defines the observed state of FybrikApplication.
-type FybrikApplicationStatus struct {
+// AssetState defines the observed state of an asset
+type AssetState struct {
```

Comment:
@shlomitk1 Would this be per data asset or per data asset flow?  (read/write/delete).  Currently seems like we have mixed them all together in a single structure, and made assumptions about how the user is supposed to figure out which flow it's for.  Ex: If CatalogAsset is populated then we assume we wrote a new asset, so I assume there shouldn't be an endpoint ... and vice versa.  I think at a minimum we should be this logic in the comments OR declare it explicitly in the structure.

Reply:
It is the question of granularity. Suppose the workload reads A, reads B, updates A. The current approach indicates that A is ready if both read and write flows including A are deployed. Do we need to tell the user "you can read A but not write" or "A can be used"?

Reply:
@simanadler I understand it should be per data asset flow. We need to change the spec, and status will be changed accordingly. Will be handled in a future PR.